client/src/pages/Project.tsx

```diff
@@ -173,6 +240,293 @@ const Project = (): JSX.Element => {
+                <div className="flex-col w-full">
+                    <input
+                        className="w-80 border-2 border-gray-200 rounded"
+                        type="text"
+                        placeholder="New Description"
+                        onChange={(e) => setNewDesc(e.target.value)}
+                    ></input>
+                    <Button type="submit" onClick={handleChangeDesc}>
+                        Change
+                    </Button>
+                </div>
+                <div className="mt-2">
+                    <Button2
+                        onClick={() => {
+                            setShowRemModal(true);
+                            setShowSettingsModal(false);
+                        }}
+                    >
+                        Edit Users
+                    </Button2>
+                    <Button2>
+                        <Link to={`/project_applications/${projectid}`}>User Join Requests</Link>
+                    </Button2>
+                </div>
+                <div className="mt-2">
+                    {isPublic ? (
+                        <Button2 type="submit" onClick={handleChangePublic}>
+                            Make Private
+                        </Button2>
+                    ) : (
+                        <Button2 type="submit" onClick={handleChangePublic}>
+                            Make Public
+                        </Button2>
+                    )}
+                    <ButtonDeleteProject
+                        onClick={() => {
+                            setShowDeleteModal(true);
+                            setShowSettingsModal(false);
+                        }}
+                    >
+                        Delete Project
+                    </ButtonDeleteProject>
+                </div>
+            </div>
+        </>
+    );
+
+    const RemoveUsersModal = (
+        <>
+            <h1 className="mb-1 text-2xl underline underline-offset-auto">Edit Users</h1>
+            <div className="w-full flex-col" key={currUsers.length}>
+                <div className="w-full ">
+                    Current Users:
+                    <div className="w-80">
+                        {currUsers.map((User) => {
+                            return (
+                                <>
+                                    <li className="flex flex-row mt-1 border-2 items-center rounded" key={User.id}>
+                                        <div className="flex justify-items-start flex-grow">{User.name}</div>
+                                        <Button
+                                            className="flex justify-end"
+                                            onClick={() => {
+                                                let tempRemovedUsers = removedUsers;
+                                                tempRemovedUsers.push(User);
+                                                setRemovedUsers(tempRemovedUsers);
```

Comment:
all these react setstates should be changed to be use callback states to prevent racing, because react states don't update immediately (can look at dataprovider.tsx, many examples there)

basically, for [s, setS], never ever use s's value directly to update the new s. Use the callback function to ensure that the value is always the latest